src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
-              // modification.
-              map.getLayers().forEach(function(item) {
-                if (gaMapUtils.isStoredKmlLayer(item)) {
-                  layer = item;
-                }
-              });
-
-              // Attach events to the source to update the KML file stored
-              // and to remove overlays when necessary
-              var source = layer.getSource();
-              unSourceEvents = [
-                source.on('addfeature', saveDebounced),
-                source.on('changefeature', saveDebounced),
-                source.on('removefeature', saveDebounced)
-              ];
-
-            }
-
-            // Attach the snap interaction to the new layer's source
-            if (snap) {
-              map.removeInteraction(snap);
-            }
-            snap = new ol.interaction.Snap({
-              source: layer.getSource()
-            });
-            map.addInteraction(snap);
+        });
 
-            // Send the choosen layer to other components
-            if (scope.options.broadcastLayer) {
-              $rootScope.$broadcast('gaDrawingLayer', layer);
-            }
-          };
-
-          // Activate the component: active a tool if one was active when draw
-          // has been deactivated.
-          var activate = function() {
-            if (!scope.options.noModify) {
-              map.addInteraction(select);
-              map.addInteraction(modify);
-            }
-            defineLayerToModify();
+        var defineLayerToModify = function() {
 
-            if (layer.adminId) {
-              updateShortenUrl(layer.adminId);
-            }
-            ol.Observable.unByKey(unLayerAdd);
-            ol.Observable.unByKey(unLayerRemove);
-            while (unWatch.length) {
-              unWatch.pop()();
-            }
-            // if a layer is added from other component (Import KML, Permalink,
-            // DnD ...) and the currentlayer has no features, we define a
-            // new layer.
-            unLayerAdd = map.getLayers().on('add', function(evt) {
-              if (gaMapUtils.isStoredKmlLayer(evt.element) &&
-                  layer.getSource().getFeatures().length == 0 &&
-                  !useTemporaryLayer) {
-                defineLayerToModify();
+          // Unregister the events attached to the previous source
+          while (unSourceEvents.length) {
+            ol.Observable.unByKey(unSourceEvents.pop());
+          }
+          ol.Observable.unByKey(unLayerVisible);
+
+          // Set the layer to modify if exist, otherwise we use an empty
+          // layer
+          layer = createDefaultLayer(map, useTemporaryLayer);
+          if (!useTemporaryLayer) {
+
+            // If there is a layer loaded from public.admin.ch, we use it for
+            // modification.
+            map.getLayers().forEach(function(item) {
+              if (gaMapUtils.isStoredKmlLayer(item)) {
+                layer = item;
```

Comment:
hmm no. The last public.geo kml layer   is modifiable, the others stay the same.